x/nft/model.go

```diff
@@ -19,12 +19,13 @@ func (m *NonFungibleToken) Validate() error {
 	if err := weave.Address(m.Owner).Validate(); err != nil {
 		return err
 	}
-	// TODO: impl proper validation
-	//for _, a := range m.ActionApprovals {
-	//if err := a.Validate(); err != nil {
-	//	return err
-	//}
-	//}
+
+	//TODO: Figure out what we do with custom action
```

Comment:
@ethanfrey @alpe 
I would like to introduce a type string field to our NFT model in order to be able to run approval validation including actions specific to token type.

I see no other good way to tell the validator to include these except passing the type.
That would also help us bulletproof the whole validation process as we are going to have switch cases for each type in approval validations, and those not implemented will throw explicit errors.

Let me know what you think.

Reply:
As I understand, you validate the list of ActionApprovals, to make sure they are legal. I don't see how much info needs to be custom here. Sure, executing the action is custom, but the approvals are quite generic.

I think I miss the problem... maybe you can explain again.